### PR TITLE
Chore/vite migration

### PR DESCRIPTION
**Pull request by mariojsnunes**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Developer experience (improves developer workflows for contributing to the project)

## Description

This PR replaces `webpack` and `CRA` in favor of `vite`.
Tried to keep `Jest` but had too many issues so migrated to `vitest`.
Had issues on all pipeline steps due to:
- commonjs/esm
- typescript errors
- functions still use webpack and commonjs, and were importing the new vite/esm packages
- observable decorators stopped working for some reason, had to migrate to the recommended approach
- loading interfaces/types from `oa-components` caused issues, so replaced with `any` as a temporary fix
- cypress configuration with vite
 
## Git Issues

Closes #

### CODE REVIEW

File: packages/cypress/src/support/db/endpoints.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { generateDBEndpoints } from 'oa-shared'
 
 // React apps populate a process variable, however it might not always be accessible outside
 // (e.g. cypress will instead use it's own env to populate a prefix)
-const process = globalThis.process || ({} as any)
+// const process = globalThis.process || ({} as any)
```
Comment (benfurber) [inline packages/cypress/src/support/db/endpoints.ts:5]: What's the context to this change?

**Review by benfurber — COMMENTED** [review]
Gone through in order up until the common folder. Had some minor comments so thought I'd post them as I need to do other things for a bit.

File: functions/webpack.config.ts
[the remark this answers is not in the document]
Comment (mariojsnunes) [reply functions/webpack.config.ts:1]: yes, functions still use webpack. can be done on a future PR

File: packages/components/src/VideoPlayer/VideoPlayer.tsx [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import ReactPlayer from 'react-player/lazy'
+import ReactPlayer from 'react-player'
```
[the remark this answers is not in the document]
Comment (mariojsnunes) [reply packages/components/src/VideoPlayer/VideoPlayer.tsx:1]: No, but lazy fails with vite. should be fixed with the 3.0 release
https://github.com/cookpete/react-player/issues/1443

File: packages/components/src/test/setup.ts [hunk 3]
```diff
@@ -1,9 +1,9 @@
-import matchers from '@testing-library/jest-dom/matchers'
+// import matchers from '@testing-library/jest-dom/matchers'
 import { cleanup } from '@testing-library/react'
-import { afterEach, expect } from 'vitest'
+import { afterEach } from 'vitest'
 
 // extends Vitest's expect method with methods from react-testing-library
-expect.extend(matchers)
+// expect.extend(matchers)
```
[the remark this answers is not in the document]
Comment (mariojsnunes) [reply packages/components/src/test/setup.ts:6]: couldn't get it to work... got this error:
image

I believe this is the reason `import '@testing-library/jest-dom/vitest'` had to be included on most test files.

File: packages/components/src/index.ts [hunk 4]
```diff
@@ -57,4 +57,5 @@ export { UserEngagementWrapper } from './UserEngagementWrapper/UserEngagementWra
 export { IconCountWithTooltip } from './IconCountWithTooltip/IconCountWithTooltip'
 export { DonationRequest } from './DonationRequest/DonationRequest'
 export { DonationRequestModal } from './DonationRequestModal/DonationRequestModal'
-export { UserNotificationItem } from './NotificationItem/NotificationItem'
+// export { IImageGalleryItem } from './ImageGallery/ImageGallery'
+// export { availableGlyphs } from './Icon/types'
```
[the remark this answers is not in the document]
Comment (mariojsnunes) [reply packages/components/src/index.ts:61]: got weird issues importing interfaces and types from the components library...
as a workaround found a way to not use them.

File: packages/components/src/TextNotification/TextNotification.tsx [hunk 5]
```diff
@@ -6,14 +6,14 @@ const fadeIn = keyframes({
   to: { opacity: 1 },
 })
 
-export interface Props {
+export interface TextNotificationProps {
```
[the remark this answers is not in the document]
Comment (mariojsnunes) [reply packages/components/src/TextNotification/TextNotification.tsx:9]: unlike others, this is exported so a better name is important.
(I'd prefer the convention to be ComponentNameProps)

File: packages/cypress/src/support/db/endpoints.ts (hunk 1 above)
Comment (mariojsnunes) [reply packages/cypress/src/support/db/endpoints.ts]: apparently isn't needed, I put it back, doesn't seem to have an effect

**Comment by davehakkens** [comment]
Can a PR like this break/change things on the frontend? 

-Asking because it seem to touch many areas and often after PR's like this many small functionalities stop working, the site is large, we dont test them all. Resulting into many bug reports and a bad user experience..

So this is me trying to be ahead of that🙃

**Comment by mariojsnunes** [comment]
Forgot to add:
- changed library that renders country flags due to esm incompatibility and it wasn't maintained for 5 years
- also due to esm, updated the Uppy version and configured it's compressor (automatically optimizes images)

File: index.html [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+      rel="preconnect"
+      href="https://firebasestorage.googleapis.com"
+      crossorigin
+    />
+    <noscript><link rel="stylesheet" href="path/to/stylesheet.css" /></noscript>
+    <meta name="theme-color" content="#000000" />
+    <meta property="og:title" content="Community Platform" />
+    <meta
+      property="og:description"
+      content="A series of tools for the Precious Plastic community to collaborate around the world. Connect, share and meet each other to tackle plastic waste."
```
[the remark this answers is not in the document]
Comment (mariojsnunes) [reply index.html:20]: As I understand it, they need to exist and are changed before the deploy. See `scripts/postBuild.ts`

**Comment by davehakkens** [comment]
When browsing 3 minutes (not going much into functionality test)
1. Blobs dont show correct
Screenshot 2024-06-04 at 10 45 12
<br>

2. profiles tabs dont show
Screenshot 2024-06-04 at 10 45 51
<br>

3. Notification got black instead of grey
Screenshot 2024-06-04 at 10 46 13
<br>

4. Patreon connection doesnt work. (could be because its devsite?)
Screenshot 2024-06-04 at 10 46 37
<br>

5. useful on how-to doesnt seem to work
Screenshot 2024-06-04 at 10 48 24


Conclusion. I think we need more real testing before merging PR's like this.

**Comment by mariojsnunes** [comment]
> **Conclusion. I think we need more real testing before merging PR's like this.**

Thanks for the tests!
Before merging I'll go over all the component changes again.

Have you tried image upload feature? Notice any quality difference?

As for the reported issues:
- I can replicate all locally except the 1.
- For 4. @benfurber can you confirm this would be due being the preview site?

**Comment by mariojsnunes** [comment]
All those reported issues are fixed. (still need to confirm the Patreon issue)

Developer notes:
- using inline svgs now requires a double quote:
before: url(path-to-svg.svg)
after: url("path-to-svg.svg")

- using decorators no longer works (cause for .5):
image
the issue was reported only for howtos, but the fix was applied to all stores.

**Comment by davehakkens** [comment]
> Have you tried image upload feature? Notice any quality difference? @mariojsnunes  

Works nice for me.


Dont see any visual errors after a quick browse.
But should all functions also be tested? (thinking that the useful button didnt work before)

**Comment by mariojsnunes** [comment]
> (thinking that the useful button didnt work before)

The cause for that was decorators used by our stores like @observable that stopped working with vite. They have now all been migrated to the new approach, so the risk is low.

**Review by benfurber — APPROVED** [review]

**Pull request merged**